libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp

```diff
@@ -89,8 +91,16 @@ AP_BattMonitor_Analog::read()
         _state.healthy &= _curr_pin_analog_source->set_pin(_curr_pin);
 
         // read current
-        _state.current_amps = (_curr_pin_analog_source->voltage_average() - _curr_amp_offset) * _curr_amp_per_volt;
 
+        if (!(uint32_t(_params._options.get()) & uint32_t(AP_BattMonitor_Params::Options::Synthetic_Current_Sensor))) {
+            _state.current_amps = (_curr_pin_analog_source->voltage_average() - _curr_amp_offset) * _curr_amp_per_volt;            
+        } 
+#if !APM_BUILD_TYPE(APM_BUILD_AP_Periph)
```

Comment:
```suggestion
#if HAL_GCS_ENABLED
```

... but really, getting the throttle back from the gcs() library?!

Reply:
Maybe `AP::vehicle().get_control_output(AP_Vehicle::ContronOutput::Throttle)`?

Reply:
Looks like only rover implemented it?